phoenix-core/src/test/java/org/apache/phoenix/util/bson/ComparisonExpressionUtilsTest.java

```diff
@@ -1954,4 +1954,272 @@ private static RawBsonDocument getDocumentValue() {
     return RawBsonDocument.parse(json);
   }
 
+  @Test
+  public void testFieldTypeFunction() {
+    RawBsonDocument rawBsonDocument = getFieldTypeTestDocument();
+    RawBsonDocument compareValues = getFieldTypeCompareValDocument();
+
+    // Test String type (S)
+    assertTrue(SQLComparisonExpressionUtils.evaluateConditionExpression(
```

Comment:
nit: how about at least one test for array element at some index?

Reply:
done